Modules/System/AL Rest Client/src/ALRestClient.Codeunit.al

```diff
@@ -0,0 +1,339 @@
+            Error(HttpResponseMessage.GetErrorMessage());
+
+        Response := HttpResponseMessage.Content().AsJson();
+    end;
+    #endregion
+
+    #region GenericSendMethods
+    /// <summary>Sends a request with the specific Http method and an empty content to the specified Uri and returns the response message.</summary>
+    /// <remarks>The function fails with an error message if the request could not be sent or a response was not received.
+    /// If a response was received, then the response message object contains information about the status.</remarks>
+    /// <param name="Method">The HTTP method to use.</param>
+    /// <param name="RequestUri">The Uri the request is sent to.</param>
+    /// <returns>The response message object</returns>
+    procedure Send(Method: Enum "Http Method"; RequestUri: Text) HttpResponseMessage: Codeunit "AL Http Response Message"
+    var
+        EmptyALHttpContent: Codeunit "AL Http Content";
+    begin
+        HttpResponseMessage := Send(Method, RequestUri, EmptyALHttpContent);
+    end;
+
+    /// <summary>Sends a request with the specific Http method and the given content and content type to the specified Uri and returns the response message.</summary>
+    /// <remarks>The function fails with an error message if the request could not be sent or a response was not received.
+    /// If a response was received, then the response message object contains information about the status.
+    /// If the content is of type Codeunit (Temp Blob) or InStream, then provide the ContentType parameter as well.</remarks>
+    /// <param name="Method">The HTTP method to use.</param>
+    /// <param name="RequestUri">The Uri the request is sent to.</param>
+    /// <param name="Content">The content to send. Valid types are: Codeunit (Temp Blob), InStream, JsonObject, JsonArray, Jsontoken, XmlDocument, Text</param>
+    /// <returns>The response message object</returns>
+    procedure Send(Method: Enum "Http Method"; RequestUri: Text; Content: Variant) HttpResponseMessage: Codeunit "AL Http Response Message"
+    var
+        ALHttpContent: Codeunit "AL Http Content";
+    begin
+        HttpResponseMessage := Send(Method, RequestUri, ALHttpContent.Create(Content));
+    end;
+
+    /// <summary>Sends a request with the specific Http method and the given content and content type to the specified Uri and returns the response message.</summary>
+    /// <remarks>The function fails with an error message if the request could not be sent or a response was not received.
+    /// If a response was received, then the response message object contains information about the status.</remarks>
+    /// <param name="Method">The HTTP method to use.</param>
+    /// <param name="RequestUri">The Uri the request is sent to.</param>
+    /// <param name="Content">The content to send. Valid types are: Codeunit (Temp Blob), InStream, JsonObject, JsonArray, Jsontoken, XmlDocument, Text</param>
```

Comment:
Here a Variant can be Codeunit "Temp Blob", but the overload suggests Codeunit "AL Http Content", so assume the explicit overload takes priority over the Variant type?
Also, the PostAsJson/PutAsJson/PatchAsJson/etc. all use overloads instead of a Variant, which seems a bit inconsistent.
I would suggest dropping the Variant type in favor of explicit types, so the compiler can help.

Reply:
Yes, agree. It was also mentioned by @IhorHandziuk, so I guess this is the way to go. I've dropped the Variant parameter in favor of explicit types.